dot/rpc/modules/grandpa.go

```diff
@@ -71,3 +97,78 @@ func (gm *GrandpaModule) ProveFinality(r *http.Request, req *ProveFinalityReques
 
 	return nil
 }
+
+// RoundState returns the state of the current best round state as well as the ongoing background rounds.
+func (gm *GrandpaModule) RoundState(r *http.Request, req *EmptyRequest, res *RoundStateResponse) error {
+	voters := gm.blockFinalityAPI.GetVoters()
+	votersPkBytes := make([]ed25519.PublicKeyBytes, len(voters))
+	for i, v := range voters {
+		votersPkBytes[i] = v.PublicKeyBytes()
+	}
+
+	prevotes, _ := gm.blockFinalityAPI.PreVotes()
+	precommits, _ := gm.blockFinalityAPI.PreCommits()
+
+	missingPrevotes, err := toAddress(difference(votersPkBytes, prevotes))
+	if err != nil {
+		return err
+	}
+
+	missingPrecommits, err := toAddress(difference(votersPkBytes, precommits))
+	if err != nil {
+		return err
+	}
```

Comment:
the equivocations would be needed to check the `difference`, since difference returns the authorities that haven't voted yet

Reply:
Done!